phase1-coordinator/src/coordinator_state.rs

```diff
@@ -2324,12 +2416,15 @@ impl CoordinatorState {
     /// On success, returns a list of justifications for the coordinator to take actions on.
     ///
     #[inline]
-    pub(super) fn update_dropped_participants(&mut self) -> Result<Vec<Justification>, CoordinatorError> {
+    pub(super) fn update_dropped_participants(
+        &mut self,
+        time: &dyn TimeSource,
+    ) -> Result<Vec<Justification>, CoordinatorError> {
         // Fetch the timeout threshold for contributors.
```

Comment:
TODO: 

- [ ] refactor this into separate methods for each type of drop
- [ ] add drop for lock timeout